lib/cpm/checks/universal/syntax-bash.sh

```
@@ -0,0 +1,34 @@
+
+source lib/cpm/shell/init.sh 2>/dev/null || true
+
+main() {
+  local failed=0 checked=0
+
+  while IFS= read -r file; do
+    [[ -z "$file" ]] && continue
+    checked=$((checked + 1))
+    local output
+    output=$(bash -n "$file" 2>&1)
+    if [[ $? -ne 0 ]]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.

(SC2181)